control-plane/p4RuntimeArchHandler.cpp

```diff
@@ -39,16 +39,20 @@ namespace ControlPlaneAPI {
 namespace Helpers {
 
 boost::optional<ExternInstance>
-getExternInstanceFromProperty(const IR::P4Table* table,
+getExternInstanceFromProperty(P4RuntimeArchHandlerIface* archHandler,
+                              const IR::TableBlock* tableBlock,
                               const cstring& propertyName,
                               ReferenceMap* refMap,
                               TypeMap* typeMap,
                               bool *isConstructedInPlace) {
+    const IR::P4Table* table = tableBlock->container;
+    CHECK_NULL(table);
+    auto tableName = archHandler->getControlPlaneName(tableBlock);
     auto property = table->properties->getProperty(propertyName);
     if (property == nullptr) return boost::none;
     if (!property->value->is<IR::ExpressionValue>()) {
         ::error("Expected %1% property value for table %2% to be an expression: %3%",
-                propertyName, table->controlPlaneName(), property);
+                propertyName, tableName, property);
```

Comment:
since you are touching all these "error" calls you may consider migrating them to our new error function prototype which has an extra argument indicating the error kind.